### DISCUSSION TITLE
Everything is ready, but still reporting an error. 'librustdesk.so' is already exists at this path '/flutter/android/app/src/main/jniLibs/arm64-v8a/librustdesk.so'

### DISCUSSION THREAD

**Discussion by LJRFox** [rustdesk/rustdesk#4210] [Q&A]
'librustdesk.so' is already exists at this path '/flutter/android/app/src/main/jniLibs/arm64-v8a/librustdesk.so', BUT  failed to load.

ps: I run the debug version manually.
 
Unhandled Exception: Invalid argument(s): Failed to load dynamic library 'librustdesk.so': dlopen failed: cannot locate symbol "vpx_codec_vp8_cx" referenced by "/data/app/~~e5v5He4rK7GwWGytVM2trA==/com.carriez.flutter_hbb-p0tgcdbY5uZTqvJrcznCkw==/lib/arm64/librustdesk.so"...

**Comment by LJRFox** [comment]
I have tried many times and it is the same, can anyone help me!

**Reply by jetwu** [reply]
Is the problem solved? thanks

**Comment by LJRFox** [comment]
![image](https://user-images.githubusercontent.com/2589262/234829490-0be018b4-b7a0-482e-8023-7b999648a9f9.png)

**Comment by deep-soft** [comment]
If you try to build android version, locally, you must replicate all steps from CI (.github/workflows/flutter-build.yml, job: build-rustdesk-android: arch: aarch64) and, before, you must do cleanup.
Perhaps some files from previous build conflict with the new ones.

After first succesfull build for Windows, I, myself, tried to do the same for android but after some fails I  resigned and discovered CI build.
After that I use only the builds from github CI.

**Reply by changcw83** [reply]
can be more specific what you mean by "replicate all steps from CI (.github/workflows/flutter-build.yml, job: build-rustdesk-android: arch: aarch64)"? And when I ran flutter/ndk_arm64.sh I got into error when compiling libsodium-sys

**Reply by jetwu** [reply]
Is the problem solved? thanks

**Reply by deep-soft** [reply]
Read here:
https://github.com/rustdesk/rustdesk/discussions/4210#discussioncomment-5742693
If you really want to build locally read all information from:
file: .github/workflows/flutter-build.yml
job: build-rustdesk-android: arch: aarch64

The error you mentioned is because you skipped one or more steps.

**Comment by b2ccoin-project** [comment]
same problem here

**Comment by ronnieCodeWang** [comment]
Has this problem been resolved?

**Reply by jetwu** [reply]
pengge nancheng third

**Comment by crowmatix** [comment]
I got exactly the same problem with librustdesk.so
